File: chess/chess.rb
count = 0.0
n = 10000000

for k in 0..99
	start = Time.now
	for i in 0..n

		knightx = rand() * 8
		knightx = knightx.to_i

		knighty = rand() * 8
		knighty = knighty.to_i

		x = rand() * 8
		x = x.to_i

		y = rand() * 8
		y = y.to_i

		if knightx + 2 == x && knighty +1 == y
			count = count + 1
		elsif knightx + 2 == x && knighty - 1 == y
			count = count + 1
		elsif (knightx - 2 == x && knighty +1 ==y)
			count = count + 1
		elsif (knightx - 2 == x && knighty -1 ==y)
			count = count + 1
		elsif knightx + 1 == x && knighty +2 ==y
			count = count + 1
		elsif knightx - 1 == x && knighty +2 ==y
			count = count + 1
		elsif knightx + 1 == x && knighty -2 ==y
			count = count + 1
		elsif knightx -1 == x && knighty -2 ==y
			count = count + 1
		end
	end
	fin = Time.now
	puts fin - start
end

File: chess/chess.c
#include <time.h>
#include <stdlib.h>
#include <stdio.h>
#include <FPT.h>

int main(){
		clock_t begin,end;
	int n = 10000000;
	int i;
	
	srand(time(NULL));
		int k;
	for(k=0;k<100;k++){
		begin = clock();
	double count = 0.0;
	for(i=0; i< n; i++){
		int knightx = (int)((rand()/(double)RAND_MAX)*8);
		int knighty = (int)((rand()/(double)RAND_MAX)*8);
		int x = (int)((rand()/(double)RAND_MAX)*8);
		int y = (int)((rand()/(double)RAND_MAX)*8);

		if(knightx + 2 == x && knighty +1 ==y){
			count++;
		}else if(knightx + 2 == x && knighty - 1 ==y){
			count++;
		}else if(knightx - 2 == x && knighty +1 ==y){
			count++;
		}else if(knightx - 2 == x && knighty -1 ==y){
			count++;
		}else if(knightx + 1 == x && knighty +2 ==y){
			count++;
		}else if(knightx - 1 == x && knighty +2 ==y){
			count++;
		}else if(knightx + 1 == x && knighty -2 ==y){
			count++;
		}else if(knightx -1 == x && knighty -2 ==y){
			count++;
		}

	}
		end = clock();

	double dur = (double)(end-begin)/CLOCKS_PER_SEC;
	printf("%lf\n", dur);
	// printf("pi = %lf\n",(((double)hit)/(double)trials) *4);
}
}